src/model/compat.cppo.ml

```diff
@@ -235,8 +235,30 @@ type 'a shape_uid_map = 'a Shape.Uid.Map.t
 type uid_to_loc = Warnings.loc Types.Uid.Tbl.t
 let empty_map = Shape.Uid.Map.empty
 
+#if OCAML_VERSION < (5,2,0)
 let shape_info_of_cmt_infos : Cmt_format.cmt_infos -> (shape * uid_to_loc) option =
  fun x -> Option.map (fun s -> (s, x.cmt_uid_to_loc)) x.cmt_impl_shape
+#else
+let loc_of_declaration =
```

Comment:
It might be good to have this function local to `shape_info_of_cmt_infos` (to avoid having a different signature for `Compat` on 5.2 and 5.1)

Reply:
Good point, done.